src/flow.ts

```diff
@@ -905,20 +987,22 @@ export class Flow {
 
     this.flags = newFlags | (this.flags & (FlowFlags.UNCHECKED_CONTEXT | FlowFlags.CTORPARAM_CONTEXT));
 
-    // local flags
+    // local flags and type
     var thisLocalFlags = this.localFlags;
     if (leftFlags & FlowFlags.TERMINATES) {
       if (!(rightFlags & FlowFlags.TERMINATES)) {
         let rightLocalFlags = right.localFlags;
         for (let i = 0, k = rightLocalFlags.length; i < k; ++i) {
           thisLocalFlags[i] = rightLocalFlags[i];
         }
+        this.narrowedTypes = right.narrowedTypes;
```

Comment:
I wonder if assigning the narrowed types map by reference here can lead to issues. Say, one of the flows is updated later on, then the other would be updated as well, which might be incorrect? Copying instead would likely be safer.

Reply:
For detail scenes to be used, copy by reference in current compile is no problem because it is only used in `TernaryExpression` and `IfStatement`. But for better scalability clone will be safer.